gadget/layout.go

```diff
@@ -243,49 +282,36 @@ func LayoutVolume(gadgetRootDir string, volume *Volume, constraints LayoutConstr
 	return vol, nil
 }
 
-// ResolveContentPaths resolves any "$kernel:" refs in the gadget
-// content and populates VolumeContent.resolvedSource with absolute
-// paths.
-//
-// XXX: maybe move into LayoutVolume(), operator on *Volume and make private?
-func ResolveContentPaths(lv *LaidOutVolume, gadgetRootDir, kernelRootDir string) error {
-	// Note that the kernelRootDir may reference the running
-	// kernel if there is a gadget update or the new kernel if
-	// there is a kernel update.
-	kernelInfo, err := kernel.ReadInfo(kernelRootDir)
-	if err != nil {
-		return err
+func resolveVolumeContent(gadgetRootDir, kernelRootDir string, kernelInfo *kernel.Info, ps *LaidOutStructure) ([]ResolvedContent, error) {
+	if !ps.HasFilesystem() {
+		// structures without a file system are not resolved here
+		return nil, nil
```

Comment:
a TODO about returning an error here?

Reply:
we have similar code for the layout case fwiw

Reply:
I am happy to change this. But I was following the pattern from layOutStructureContent() which is doing the same.

Reply:
Then it's fine I guess. We'll need to tweak this and the loop below anyway in a PR returning an error for $kernel in bare content